modules/trustxBidAdapter.js

```diff
@@ -1,165 +1,147 @@
+const BIDDER_CODE = 'trustx';
+const ENDPOINT_URL = '//sofia.trustx.org/hb';
+const TIME_TO_LIVE = 360;
+const ADAPTER_SYNC_URL = '//sofia.trustx.org/push_sync';
+const LOG_ERROR_MESS = {
+  noAuid: 'Bid from response has no auid parameter - ',
+  noAdm: 'Bid from response has no adm parameter - ',
+  noBid: 'Array of bid objects is empty',
+  noPlacementCode: 'Can\'t find in requested bids the bid with auid - ',
+  emptyUids: 'Uids should be not empty',
+  emptySeatbid: 'Seatbid array from response has empty item',
+  emptyResponse: 'Response is empty',
+  hasEmptySeatbidArray: 'Response has empty seatbid array',
+  hasNoArrayOfBids: 'Seatbid from response has no array of bid objects - '
+};
+const bidsRequestMap = {};
+export const spec = {
+  code: BIDDER_CODE,
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.uid && bid.adUnitCode);
+  },
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} validBidRequests - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests) {
+    const auids = [];
+    const bidsMap = {};
+    const bids = validBidRequests || [];
+    const bidderRequestId = bids[0] && bids[0].bidderRequestId;
+
+    bids.forEach(bid => {
+      if (!bidsMap[bid.params.uid]) {
+        bidsMap[bid.params.uid] = [bid];
+        auids.push(bid.params.uid);
+      } else {
+        bidsMap[bid.params.uid].push(bid);
+      }
+    });
 
-var TrustxAdapter = function TrustxAdapter() {
-  const bidderCode = 'trustx';
-  const reqHost = '//sofia.trustx.org';
-  const reqPath = '/hb?';
-  const LOG_ERROR_MESS = {
-    noAuid: 'Bid from response has no auid parameter - ',
-    noAdm: 'Bid from response has no adm parameter - ',
-    noBid: 'Array of bid objects is empty',
-    noPlacementCode: 'Can\'t find placementCode for bid with auid - ',
-    havePCodeFor: ', placementCode is available only for the following uids - ',
-    emptyUids: 'Uids should be not empty',
-    emptySeatbid: 'Seatbid array from response has empty item',
-    emptyResponse: 'Response is empty',
-    hasEmptySeatbidArray: 'Response has empty seatbid array',
-    hasNoArrayOfBids: 'Seatbid from response has no array of bid objects - '
-  };
+    bidsRequestMap[bidderRequestId] = bidsMap;
 
-  function _makeHandler(auids, placementMap) {
-    var cbName = bidderCode + '_callback_wrapper_' + auids.join('_');
-    $$PREBID_GLOBAL$$[cbName] = function(resp) {
-      delete $$PREBID_GLOBAL$$[cbName];
-      _responseProcessing(resp, auids, placementMap);
+    const payload = {
+      u: utils.getTopWindowUrl(),
+      pt: window.globalPrebidTrustxPriceType === 'gross' ? 'gross' : 'net',
```

Comment:
you can pass `window.globalPrebidTrustxPriceType` via param. This is not allowed.